### DISCUSSION TITLE
[blocks]: Time Picker and Time Range Picker

### DISCUSSION THREAD

**Discussion by HenriqueBarucco** [shadcn-ui/ui#3174] [Blocks Request]
### Description

Description
A simple time picker (just as date picker, but only for time) and also a time range picker.

love the work 👍🏻

Example
[Mobiscroll Time Range Component](https://demo.mobiscroll.com/react/range/#)

### Example

https://demo.mobiscroll.com/react/range/#

**Comment by birobirobiro** [comment]
Time picker: https://time.openstatus.dev/

Date range picker: https://date-range-picker-for-shadcn-demo.vercel.app/

**Reply by Jbmanllr** [reply]
Real time picker here: https://github.com/fatihcaliss/shadcn-time-range-picker, above is a time input.

**Comment by jt6677** [comment]
nice! would love to see a time range (from and to) with validation.

**Comment by jt6677** [comment]
import { isAfter } from 'date-fns'

 const [fromDate, setFromDate] = useState<Date | undefined>(new Date())
  const [toDate, setToDate] = useState<Date | undefined>(new Date())
  const [isValid, setIsValid] = useState(false)

  useEffect(() => {
    if (fromDate && toDate) {
      setIsValid(isAfter(toDate, fromDate))
    }
  }, [fromDate, toDate])        is quite easy with date-fn. could probably wired it up with <Form> to make it show the red error in label.

**Comment by fatihcaliss** [comment]
Hello, I created this package, you can look over.
GitHub: [https://github.com/fatihcaliss/shadcn-time-range-picker](https://github.com/fatihcaliss/shadcn-time-range-picker)
npm: [https://www.npmjs.com/package/shadcn-time-range-picker](https://www.npmjs.com/package/shadcn-time-range-picker)

**Reply by Jbmanllr** [reply]
Thanks as a "time picker" it's a better option than the time input above!
